MAID-2727: minor fixes

povilasb opened the pull request:
These are the changes required to integrate safe_crypto into Routing.

povilasb committed:
feat/lib: expose constants for key sizes

If I want to construct keys from bytes, I need the size of key buffers.

povilasb committed:
feat/lib: make some types copyable

Makes it easier to use the public keys and signature later and the wrapped types are copyable anyway.

Fraser999 commented on src/lib.rs:
```diff
@@ -97,13 +97,20 @@ use serde::{de::DeserializeOwned, Serialize};
-const SYMMETRIC_KEY_BYTES: usize = secretbox::KEYBYTES;
+/// Public encryption key length in bytes.
+pub const PUBLIC_ENCRYPT_KEY_BYTES: usize = box_::PUBLICKEYBYTES;
+/// Private encryption key length in bytes.
```
I think we should stick to "secret" rather than "private" throughout (applies to line 106 too).

Fraser999 commented on src/lib.rs:
```diff
@@ -97,13 +97,20 @@ use serde::{de::DeserializeOwned, Serialize};
+pub const PUBLIC_ENCRYPT_KEY_BYTES: usize = box_::PUBLICKEYBYTES;
+/// Private encryption key length in bytes.
+pub const SECRET_ENCRYPT_KEY_BYTES: usize = box_::SECRETKEYBYTES;
+/// Public sign key length in bytes.
```
I guess to be exact, it should be "signing" rather than "sign" (applies to line 106 too).

povilasb committed:
chore/travis: run clippy on crypto mock too

povilasb committed:
fix/mock: make mock keys length the same as real ones

Otherwise clippy shows some 'pass by value' warnings which are not present in production code.

Fraser999 commented on src/mock_crypto.rs:
```diff
@@ -55,16 +55,16 @@ pub(crate) mod crypto_impl {
+            pub(crate) const PUBLICKEYBYTES: usize = 32;
             /// Number of bytes in a `SecretKey`.
-            pub(crate) const SECRETKEYBYTES: usize = 8;
+            pub(crate) const SECRETKEYBYTES: usize = 32;
```
The [signing secret key](https://docs.rs/rust_sodium-sys/0.10.0/rust_sodium_sys/constant.crypto_sign_ed25519_SECRETKEYBYTES.html) and [signature](https://docs.rs/rust_sodium-sys/0.10.0/rust_sodium_sys/constant.crypto_sign_ed25519_BYTES.html) should be 64 bytes each

Fraser999 commented on src/mock_crypto.rs:
```diff
@@ -107,14 +107,14 @@ pub(crate) mod crypto_impl {
                 let mut temp = m.to_vec();
                 temp.extend(&sk.0);
-                Signature(hash64(&temp))
+                Signature(hashing_impl::sha3_256(&temp))
```
To keep the speed up (and since this will need adjusted anyway if the signature length is increased to 64), we could maybe just put the result of `hash64` into the first 8 bytes and make the remainder all `0`, e.g.
```rust
let mut hash = hash64(&temp).to_vec();
hash.extend([0; 56].iter());
Signature(hash)
```
